### ISSUE TITLE
proton client ignore ctrl+c on select from system.numbers

### ISSUE THREAD

**Issue by mosinnik** [timeplus-io/proton#239]
1. open proton-client
2. query without limit `select number from system.numbers where number % 10 = 0` 
3. try pass ctrl+c

In very rare cases ctrl+c work and cause stop but usually client just ignore it.
I think that cause is that select from numbers without limit is very intensive for CPU and client ignore input stream.

So need make to react to ctrl+c as stop output action even on intensive queries

**Comment by chenziliang** [comment]
Thanks for report this.

**Comment by yokofly** [comment]
hi @mosinnik thank you for taking the trial.
it can reproduce easily (with commit 0e183fb85b36b739a5c499ea426f94be657aea66)
for the potential race, it may be. but the client display shows only consumption (1.1 CPU). 

I will take a look. thanks!

**Comment by yokofly** [comment]
@mosinnik  apologies for the late reply. We have not focused too much on proton-client previously. I spent some time comparing current proton-client vulnerabilities with clickhouse-client, and have tried to address this issue.

However, the official release version will need to be published later. I can trigger a CI action to propose a docker image or my personal local build binary for you if that would help in the meantime? 

Just wanted to do a final check - is the pic/video below accurate regarding this issue? Please let me know if any clarification or additional details would be helpful. I'm happy to assist further.

![image](https://github.com/timeplus-io/proton/assets/113408135/4a7fe3cb-92b4-473c-b286-2ad99c76b4ce)
![image](https://github.com/timeplus-io/proton/assets/113408135/221528ab-cc11-49ba-a0f0-4caadc182ce8)

https://harvest-vegetarian-745.notion.site/ctrl-c-cannot-correct-cancel-query-49898eb3bb2c4543a1360574904d7020?pvs=4

**Comment by mosinnik** [comment]
@yokofly looks very nice, as for me now it's not an urgent problem so I'll wait release

**Comment by yokofly** [comment]
> @yokofly looks very nice, as for me now it's not an urgent problem so I'll wait release

Hi @mosinnik  thanks for the patience :)
wait, umm the above picture or video is my finding problem, is my understanding of the issue accurate?

**Comment by mosinnik** [comment]
oh, may be misunderstanding. Second screen looks like my situation - query is running, stats printing but ignore ctrl+c. But on video all is going so fast so I though that is ok and already fast reacted on ctrl+c.

**Comment by yokofly** [comment]
thank you! I think this new release will solve it :)

**Comment by yokofly** [comment]
_edit v1: update `docker stop` and add `docker --rm` in launch image _

Hi, @mosinnik !
we have published the 1.3.22 version, either through docker/ single binary /homebrew.
you can try it in your free time.
 
docker: 
```
docker run -d --rm --pull always --name proton ghcr.io/timeplus-io/proton:latest
docker exec -it proton proton-client -n

select number from system.numbers where number % 10 = 0

docker stop proton
```

single binary: 
```
curl -sSf https://raw.githubusercontent.com/timeplus-io/proton/develop/install.sh | sh
./proton server start

./proton client
```

homebrew: 
```
brew tap timeplus-io/timeplus
brew install proton
```
